### PR TITLE
Add JSON session support for SwitchUserGrantedAuthority

### PR DESCRIPTION
**Pull request by markusheiden**

### CODE REVIEW

**Comment by markusheiden** [comment]
I don't know why unrelated tests fail. Hopefully this is mergeable though.

**Comment by sjohnr** [comment]
@markusheiden, thanks for the contribution!

> I don't know why unrelated tests fail.

If you want to rebase on `main`, I believe all unrelated tests should pass now.

>  Hopefully this is mergeable though.

Do you have an issue you're resolving with this PR? I don't see a related issue. If not, can you provide some context around the issue that led to submitting the PR?

**Comment by markusheiden** [comment]
I thought it was obvious what this PR wants to achieve. 
Here the corresponding issue #11775.

**Comment by sjohnr** [comment]
@markusheiden thanks for opening the issue. Note that it's usually best to open an issue before submitting a PR to allow discussion prior to spending time on the PR. I'll add additional comments on the issue.

**Comment by sjohnr** [comment]
Thanks @markusheiden, this is now merged for 6.3 as 5c5503924b8de1c507c3b485ec1ed6c17277b110 with polish commit bd345fb2a8e6f9269583c5a278a8d8f4c27ff3ee.

File: web/src/main/java/org/springframework/security/web/jackson2/WebServletJackson2Module.java [hunk 1]
```diff
@@ -56,6 +57,7 @@ public void setupModule(SetupContext context) {
 		context.setMixInAnnotations(SavedCookie.class, SavedCookieMixin.class);
 		context.setMixInAnnotations(DefaultSavedRequest.class, DefaultSavedRequestMixin.class);
 		context.setMixInAnnotations(WebAuthenticationDetails.class, WebAuthenticationDetailsMixin.class);
+		context.setMixInAnnotations(SwitchUserGrantedAuthority.class, SwitchUserGrantedAuthorityMixIn.class);
```
Comment (JohnNiang) [inline web/src/main/java/org/springframework/security/web/jackson2/WebServletJackson2Module.java:60]: Did we forget to set up this mixin for [WebServerJackson2Module](https://github.com/spring-projects/spring-security/blob/df640f22dcf63fdd1e54138cf4a358e268f7572f/web/src/main/java/org/springframework/security/web/server/jackson2/WebServerJackson2Module.java#L53-L56)?

**Pull request closed without merging**